Provide override option when import generic assay data (for review)

dippindots opened the pull request:
Provide override option when import generic assay data

related [document](https://docs.google.com/document/d/1f3JFG7tPiYXcQsJr5GHRzOK-ko_epBhOjDJa1ppgdsA/edit?disco=AAAADg3rhvo) updated here.

Changes in this pr:
1. add Provide override option (`updateInfo`) when import generic assay data.
2. log the import results after import
3. Add one more method in DaoGenericAssay to support update of the entities.

kalletlak committed:
Add new api for study view genomic data chart

dippindots committed:
update generic assay importer

pvannierop commented on core/src/main/java/org/mskcc/cbio/portal/scripts/ImportProfileData.java:
```diff
@@ -100,7 +100,7 @@ public void run() {
             } else if (geneticProfile.getGeneticAlterationType() == GeneticAlterationType.GENERIC_ASSAY) {
                 // add all missing `genetic_entities` for this assay to the database
-                ImportGenericAssayEntity.importData(dataFile, geneticProfile.getGeneticAlterationType(), geneticProfile.getOtherMetaDataField("generic_entity_meta_properties"));
+                ImportGenericAssayEntity.importData(dataFile, geneticProfile.getGeneticAlterationType(), geneticProfile.getOtherMetaDataField("generic_entity_meta_properties"), true);
```
Why do we always overwrite the entities when importing profiles? Should this not be linked to the CLI variable?

pvannierop reviewed (commented):
I am worried by a single hard-coded `true` in the profile import.

Also, should there not be an update of the importer python script that is related to this new parameter?

dippindots committed:
update generic assay importer (python script)

kalletlak committed:
Enrichments: Fix profiled cases count for groups with no alterations

inodb committed:
Fix compile error due to incorrectly sovled merge

see: daf21d0b664fa5c92dfc67c509d541d79d89b42c

inodb committed:
fix test (uppercase conflict)

inodb committed:
Merge remote-tracking branch 'origin/release-3.2.0' into rc

inodb committed:
frontend v3.2.0

inodb committed:
remove release-3.2.0 from README

inodb committed:
clean up branch table README

inodb committed:
Automated release-3.2.0 to rc merge

averyniceday committed:
Add a check for existing "SV Genetic Profile" during import
Previously, SV genetic profile was automatically generated for fusion files,
but this could result in adding multiple genetic profiles with same names
causing an exception
Added unit tests for "split" fusions file

kalletlak committed:
Filter enrichments with no mapped gene object

rmadupuri committed:
Added all sample zscore profiles

sheridancbio committed:
Merge pull request #7032 from averyniceday/fix-dupe-sv-geneticprofile

Add a check for existing "SV Genetic Profile" during import

inodb committed:
Merge pull request #7033 from cBioPortal/new_genomic_profiles

Extend valid mRNA expression profile ids

inodb committed:
Automated master to rc merge

tmazor committed:
update references to tcga provisional

tmazor committed:
more faq updates

tmazor committed:
more faq update

adamabeshouse committed:
Merge pull request #7030 from kalletlak/fix-7023

Filter enrichments with no mapped gene object

inodb committed:
Automated master to rc merge

jjgao committed:
update references to tcga provisional (#7035)

update references to tcga provisional

inodb committed:
Automated master to rc merge

inodb committed:
fix closing tag keycloak docs

onursumer committed:
added oncokb auth doc (related to #7053)

inodb committed:
Automated master to rc merge

onursumer committed:
fixed OncoKB proxy return type, and updated default url

alisman committed:
Merge pull request #7082 from onursumer/oncokb-proxy-fix

Fixed OncoKB proxy return type, and updated default URL

inodb committed:
Add news for 3.2.1

inodb committed:
v3.2.1

inodb committed:
fix year news

inodb committed:
Merge pull request #7085 from cBioPortal/news-3.2.1

Add news for 3.2.1

inodb committed:
Automated master to rc merge

rmadupuri committed:
reverted the zscore documentation

kalletlak committed:
SV: check if all values are integer before calculating bins

jjgao committed:
reverted the zscore documentation (#7100)

reverted the zscore documentation

Luke-Sikina committed:
Add Fields to config_service.jsp
- Forgot to add new fields for OQL query limit and
persistent cache to config_service.jsp

alisman committed:
Merge pull request #7103 from Luke-Sikina/add-new-fields-to-conf

Add Fields to config_service.jsp

inodb committed:
Automated master to rc merge

Commit:
Adjust spring repo URL

inodb committed:
Merge pull request #7107 from thehyve/fix_spring_url

Adjust spring repo URL

inodb committed:
Automated master to rc merge

averyniceday committed:
Performance optimization for ProfiledCasesCounter (#7088)

* Performance optimization for ProfiledCasesCounter
- Original code loops through genes and calculates gene panel cases each time
- Cache/store gene panel cases and reuse instead of recalculating
- Other cleanup related to creating maps to store reused objects
- Precompute gene panel cases prior to gene loop
- Instead of partitioning GenePanelData records first into gene panel and later constructing unique case string set ... directly compute set and store in map

Co-authored by: Avery Wang <18199796+averyniceday@users.noreply.github.com>
Co-authored-by: Robert Sheridan <7747489+sheridancbio@users.noreply.github.com>

inodb committed:
Automated master to rc merge

inodb committed:
Merge pull request #7101 from kalletlak/fix-7038

SV: check if all values are integer before calculating bins

yichaoS committed:
Public release Jan 28 2020 (#7069)

Added two new UTUC studies

Co-authored-by: Ino de Bruijn <ino@ino.bio>

inodb committed:
add-news-3.2.2

inodb committed:
change wording slightly

inodb committed:
v3.2.2

inodb committed:
change wording slightly

inodb committed:
change performance section

inodb committed:
add mrna expression

inodb committed:
Merge pull request #7126 from cBioPortal/news-3.2.2

Add news item v3.2.2

inodb committed:
attempt to fix news formatting

inodb committed:
fix formatting news again

always get reviews & test in dev environment first 😩

inodb committed:
add gif for v3.2.2 news item

inodb committed:
Automated master to rc merge

inodb committed:
add link to query for v3.2.2 news

Commit:
fix mouse gene info parsing

Commit:
correct regex

ecerami committed:
Fixed broken links;  this fixes #7115

inodb committed:
Automated master to rc merge

inodb committed:
Merge pull request #7131 from ecerami/contributing-links

Fixed broken links within CONTRIBUTING.md

inodb committed:
Automated master to rc merge

ngocnn1104 committed:
Added property into config_service to display geneset feature on frontend

kalletlak committed:
Study view: Fix error while calculating databin NA counts

oplantalech committed:
Add documentation of skin.show_gsva property

kalletlak committed:
Fix study view fusion gene filter

alisman committed:
Merge pull request #7140 from cBioPortal/config_service_geneset

Added `skin.show_gsva` to service_config

sheridancbio committed:
Merge pull request #7036 from kalletlak/fix-7010

Enrichments: Fix profiled cases count for groups with no alterations

inodb committed:
Automated master to rc merge

kalletlak committed:
Study-view:API to support all molecular profiles

inodb committed:
Merge pull request #7129 from thehyve/mouse_gene_parsing_fix

fix mouse gene info parsing

ecerami committed:
Added first draft of software acknowledgements page

inodb committed:
Merge pull request #7163 from ecerami/software-page

Added first draft of software acknowledgements page

inodb committed:
Automated master to rc merge

yichaoS committed:
New Data public release Feb 12 2020 (#7150)


Co-authored-by: Ino de Bruijn <ino@ino.bio>

ecerami committed:
Increase heading levels to fix #6186

inodb committed:
Fix web-and-data docker image build

Update libmysql dependency to be compatible with latest openjdk 8-jre
image. Seems like that image was updated and libmysql-java is no longer
available thru apt-get.

Co-authored-by: Angelica Ochoa <15623749+ao508@users.noreply.github.com>

inodb committed:
Merge pull request #7166 from ecerami/faq-fix

Increase heading levels to fix FAQ header bugs

inodb committed:
Automated master to rc merge

oplantalech committed:
Allow dots in id validation (for gene sets)

Commit:
Fix vertical bar parsing

inodb committed:
Merge pull request #7169 from thehyve/mouse_gene_parsing_fix

Fix mouse gene importer parsing (vertical bar separator)

inodb committed:
v3.2.3 🚀

Luke-Sikina committed:
Reference app.name correctly
- app.name was missing a closing braket
- this meant the cache was not being hit

inodb committed:
Automated master to rc merge

rmadupuri committed:
allow NUMERIC attribites to start with > and < symbols

zhx828 committed:
Fix the oncokb proxy issue (#7176)

* Do not return ResponseEntity

Return the body as String.
The suspicion is that the Object.class cannot convert the data properly

* Update proper http code for oncokb proxy

rmadupuri committed:
allow >,< in first position and check if the rest is number

inodb committed:
Merge pull request #7144 from Luke-Sikina/reference-app-name-correctly

Reference app.name correctly

inodb committed:
Automated master to rc merge

zhx828 committed:
Merge pull request #7143 from kalletlak/fix-7142

Study view: Fix error while calculating databin NA counts

kalletlak committed:
Fix error while fetching group with StudyViewFilter

inodb committed:
Automated master to rc merge

zhx828 committed:
Merge pull request #7070 from onursumer/oncokb-auth-doc

Added OncoKB authentication document

onursumer committed:
added version information to the OncoKB data access doc

inodb committed:
Automated master to rc merge

inodb committed:
Merge pull request #7206 from onursumer/oncokb-aut-doc-version

add version information to the OncoKB data access doc

inodb committed:
frontend v3.2.4 + 2 fixes (#7209)

frontend v3.2.4 + 2 fixes

Update frontend to a few commits after v3.2.4 which includes the fixes to

- build on jitpack (react-table typing issue) (cBioPortal/cbioportal-frontend#3078)
- bug fix related to oncokb anotation (cBioPortal/cbioportal-frontend#3079)

Co-authored-by: Hongxin <zhx828@users.noreply.github.com>

jjgao committed:
Merge pull request #7197 from cBioPortal/validator_age_check

Allow the validator to pass for NUMERIC attributes starting with > and < symbols

kalletlak committed:
Merge pull request #7204 from kalletlak/fix-7188

Fix error while fetching group with StudyViewFilter

kalletlak committed:
Merge pull request #7148 from kalletlak/fix-7145

Fix study view fusion gene filter

inodb committed:
Automated master to rc merge

yichaoS committed:
zscore doc update: method part
zscore doc update: added running command examples
zscore doc update: added data example
zscore doc update: added link to input exp file
zscore doc update: moved illustration of using -l to paramters section
z-score doc update: added original exp in data exmaple for new zscore profile
z-score doc update: update format
reviewed and separated the method contents

tmazor committed:
add mutation exclusion to OQL documentation

kalletlak committed:
Update ProfiledCasesCounter.java

kalletlak committed:
Merge pull request #7225 from kalletlak/fix-7211

Update ProfiledCasesCounter.java

inodb committed:
Automated master to rc merge

tmazor committed:
update to include mutation position exclusion

yichaoS committed:
Update URL (study_id) for prad cdk12 study

yichaoS committed:
Added missing space

Luke-Sikina committed:
Bugfix: Sample Quicksearch broke on string of " "
- Trimmed the string
- Added a test

adamabeshouse committed:
Merge pull request #7216 from cBioPortal/tmazor-patch-3

add mutation exclusion to OQL documentation

oplantalech committed:
Add option to remove studies giving study Id (#7141)

* Add option to remove studies giving study Id

Co-authored-by: Pim van Nierop <pim@thehyve.nl>

Commit:
move treatment format to generic assay

alisman committed:
Release v3.2.5

alisman committed:
Merge pull request #7248 from alisman/releaes-3-2-5

Release v3.2.5!

ritikakundra committed:
Merge pull request #7244 from cBioPortal/yichaoS-patch-3

Update URL (study_id) for prad cdk12 study

inodb committed:
Automated master to rc merge

ecerami committed:
Fix #6642 Internal Links within the File Formats Markdown Page

kalletlak committed:
Merge pull request #7245 from Luke-Sikina/fix-quick-search-for-string-of-spaces

Bugfix: Sample Quicksearch broke on string of " "

alisman committed:
Add appopriate content type attributes to script tags for frontend javascript assets

inodb committed:
Automated master to rc merge

inodb committed:
Merge pull request #7252 from ecerami/file-format-doc-fix

Fix #6642 Internal Links within the File Formats Markdown Page

rmadupuri committed:
Separated the two methods and added links to scripts

rmadupuri committed:
Added line break

alisman committed:
Merge pull request #7256 from alisman/fixScript

Add appropriate content type attributes to script tags for frontend js

dippindots committed:
update resources schema

rmadupuri committed:
Updated introduction to use high/low rather than over/under-expressed

inodb committed:
Automated master to rc merge

kalletlak committed:
Add resource API's

kalletlak committed:
Clinical Enrichment: Exclude NA category while calculating p-value

jjgao committed:
Create stale.yml

inodb committed:
frontend v3.2.8 🚀

adamabeshouse committed:
Merge pull request #7271 from kalletlak/fix-7267

Clinical Enrichment: Exclude NA category while calculating p-value

inodb committed:
Automated master to rc merge

n1zea144 committed:
Merge pull request #7247 from dippindots/update-doc

Update generic assay file format document

dippindots committed:
Merge pull request #7259 from dippindots/resources_schema

updating database schema for resource feature

inodb committed:
Automated master to rc merge

kalletlak committed:
Remove Model for ChartSetting

inodb committed:
Merge pull request #7272 from cBioPortal/jjgao-patch-1

Create stale.yml

ritikakundra committed:
Merge pull request #7215 from cBioPortal/zscore_doc_update

zscore doc update: two profiles

inodb committed:
Automated master to rc merge

dippindots committed:
resource importer implementation

jjgao committed:
limit 1 action per hour / 24 per day

inodb committed:
Merge pull request #7291 from cBioPortal/jjgao-patch-2

limit 1 Stale action per hour / 24 per day

inodb committed:
Automated master to rc merge

inodb committed:
frontend v3.2.9 🚀

inodb committed:
increase timeout localdb test circleci

inodb committed:
Merge pull request #7296 from cBioPortal/change-cirlceci-timeout-localdb

increase timeout localdb test circleci

inodb committed:
Automated master to rc merge

inodb committed:
add pathways tutorial

inodb committed:
Add news pathwaymapper

inodb committed:
Merge pull request #7205 from thehyve/fix_gene_sets_validation

Allow dots in id validation (for gene sets)

kalletlak committed:
Fix error while fetching multiple molecular profile samples

inodb committed:
change wording

this makes it more clear no new pathways can be added

inodb committed:
Automated master to rc merge

kalletlak committed:
Merge pull request #7301 from kalletlak/fix-7299

Fix error while fetching multiple molecular profile samples

inodb committed:
Merge pull request #7300 from cBioPortal/pathwaymapper-news

Add news pathwaymapper

jjgao committed:
Update Z-Score-normalization-script.md

jjgao committed:
Merge pull request #7307 from cBioPortal/jjgao-patch-3

Update Z-Score-normalization-script.md

yichaoS committed:
Mar 18, new study release

- breast_alpelisib_2020
- glioma_mskcc_2019
- mixed_cfdna_msk_2019

inodb committed:
Automated master to rc merge

alisman committed:
frontend v3.2.10 🚀

alisman committed:
Merge pull request #7310 from alisman/release-3.2.10

frontend v3.2.10 🚀

inodb committed:
Merge remote-tracking branch 'origin/master' into rc

kalletlak committed:
Merge pull request #7260 from kalletlak/fix-7236

Add resource API's

amirinenik committed:
saml local logout property added to server config

dippindots committed:
Merge pull request #7270 from dippindots/resources_importer

Resources importer implementation

inodb committed:
Merge pull request #7329 from cBioPortal/feature-resources

Feature resources

oplantalech committed:
Customize citation field

alisman committed:
Merge pull request #7330 from thehyve/customize_citation

Customize citation field

yichaoS committed:
Update study ID in URL for cfDNA study

jjgao committed:
Merge pull request #7309 from cBioPortal/yichaoS-patch-5

Mar 18, new study release

inodb committed:
Automated master to release-3.3.0 merge

inodb committed:
frontend v3.2.11

inodb committed:
Merge pull request #7226 from kalletlak/fix-6771

Study-view:API to support all molecular profiles

amirinenik committed:
Merge branch 'master' into saml-logout-local-fix

inodb committed:
Automated master to release-3.3.0 merge

sheridancbio committed:
match tested function name in unit test

n1zea144 committed:
Merge pull request #7321 from amirineni1/saml-logout-local-fix

saml local logout property added to server config

inodb committed:
Automated master to release-3.3.0 merge

leexgh committed:
add genome nexus grch38 to config

onursumer committed:
do not end with non-integer power of 10 if possible

inodb committed:
frontend v3.2.12

inodb committed:
Merge pull request #7338 from sheridancbio/adjust-persistence-unit-test-name

match tested function name in unit test

inodb committed:
Merge pull request #7353 from onursumer/log-scale-data-binner-fix

Do not end with non-integer power of 10 if possible

inodb committed:
Automated master to release-3.3.0 merge

jjgao committed:
Update News.md

schultzn committed:
Update News.md

jjgao committed:
Merge pull request #7357 from cBioPortal/jjgao-patch-3

Update News.md

inodb committed:
Automated master to release-3.3.0 merge

kalletlak committed:
Fix error while updating frontend resource client from the api

kalletlak committed:
Merge pull request #7366 from kalletlak/resource-data

Fix error while updating frontend resource client from the api

dippindots committed:
fix resource data validator bug (#7352)

update resource data validator

leexgh committed:
add property show.mutation_mappert_tool.grch38

inodb committed:
Merge pull request #7348 from leexgh/hg38

add genome nexus grch38 to config

jjgao committed:
Update DaoGeneOptimized.java

alisman committed:
Merge pull request #6643 from kalletlak/fix-6278

SV: add new api to support genomic data chart

inodb committed:
Fix test & update frontend v3.2.13

Somehow it seems like one test was testing a different method then
intended? Was giving compilation failure:

```
[ERROR] Failed to execute goal
org.apache.maven.plugins:maven-compiler-plugin:3.5.1:testCompile
(default-testCompile) on project web: Compilation failure
[ERROR]
/home/travis/build/cBioPortal/cbioportal/web/src/test/java/org/cbioportal/web/util/DataBinnerTest.java:[1252,44]
method calculateClinicalDataBins in class
org.cbioportal.web.util.DataBinner cannot be applied to given types;
[ERROR]   required:
T,org.cbioportal.web.parameter.ClinicalDataType,java.util.List<org.cbioportal.model.ClinicalData>,java.util.List<org.cbioportal.model.ClinicalData>,java.util.List<java.lang.String>,java.util.List<java.lang.String>
[ERROR]   found:
org.cbioportal.web.parameter.ClinicalDataBinFilter,org.cbioportal.web.parameter.ClinicalDataType,java.util.List<org.cbioportal.model.ClinicalData>,java.util.List<java.lang.String>
[ERROR]   reason: cannot infer type-variable(s) T
[ERROR]     (actual and formal argument lists differ in length)
[ERROR]
[ERROR] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to
execute goal
org.apache.maven.plugins:maven-compiler-plugin:3.5.1:testCompile
(default-testCompile) on project web: Compilation failure
```

inodb committed:
Automated master to release-3.3.0 merge

inodb committed:
Add news item v3.2.13

inodb committed:
address comments

inodb committed:
fix border screenshot

inodb committed:
remove custom group example

holtgrewe committed:
Allow specifying the entityBaseURL.

jjgao committed:
Merge pull request #7390 from cBioPortal/update-news-with-3.2.13

Add news item v3.2.13

leexgh committed:
fix oncokb proxy controller

inodb committed:
Automated master to release-3.3.0 merge

inodb committed:
Merge pull request #7400 from leexgh/missing-oncokb-token

fix oncokb proxy controller

kalletlak committed:
Add case list sample counts api in study view

jjgao committed:
Update News.md

inodb committed:
frontend v3.2.14

inodb committed:
Automated master to release-3.3.0 merge

holtgrewe committed:
Further documenting "#{null}" value.

Co-Authored-By: Ino de Bruijn <ino@ino.bio>

inodb committed:
Merge pull request #7399 from holtgrewe/saml-entity-url

Allow specifying the entityBaseURL.

kalletlak committed:
Coexpression: Filter sample having data for both molecular profiles

agandolf committed:
Build dev docker image for master and rc using github actions (#7410)

* Github action to perform cached docker builds

* Action to use caching and secrets

* Adds missing context

* Adjust context

* Context point to root

* Hard code root dir

* test run

* test run 2

* test run 3

* test run 4

* test run 5

* test run 6

* Adds second dockerfile build

* test run 7

* test run 8

* test run 9

* test run 10

* test run 11

* test run 12

* test run 13

* Run on push to rc/master

* Changes image name to match primary repo

mdhvprasad committed:
Make metaImport.py use subcommands instead of -c (#7401)

inodb committed:
Automated master to release-3.3.0 merge

inodb committed:
build release images using github actions

Add release-*

inodb committed:
Automated master to release-3.3.0 merge

inodb committed:
Update Customizing-your-instance-of-cBioPortal.md

tmazor committed:
update tutorial 2

inodb committed:
Merge pull request #7425 from cBioPortal/tmazor-patch-4

update tutorial 2

kalletlak committed:
Merge pull request #7409 from kalletlak/fix-7403

Coexpression: Filter sample having data for both molecular profiles

kalletlak committed:
CoExpression: fix error while querying for geneset molecular profile

Spenca committed:
Update example_commands.md

Add study deletion

inodb committed:
Automated master to release-3.3.0 merge

kalletlak committed:
Merge pull request #7429 from kalletlak/fix-coexpression

CoExpression: fix error while querying for geneset molecular profile

inodb committed:
Merge pull request #7420 from kalletlak/fix-7358

Add case list sample counts api in study view

inodb committed:
Merge pull request #7431 from Spenca/patch-2

Update example_commands.md

inodb committed:
Merge pull request #7414 from cBioPortal/remove-custom-header-from-docs

Remove skin.custom_header from docs

inodb committed:
frontend v3.2.15

inodb committed:
Merge pull request #7435 from cBioPortal/release-3.3.0

Release 3.3.0

inodb committed:
frontend v3.3.0

inodb committed:
add v3.3.0 news item

jjgao committed:
Update News.md

jjgao committed:
Merge pull request #7438 from cBioPortal/add-v3.3.0-news-item

add v3.3.0 news item

jjgao committed:
Update DaoGeneOptimized.java

kalletlak committed:
Temporary fix for error in mutation enrichments

averyniceday committed:
Update Deployment Docs (#7439)

* Update Deployment-Procedure.md

jjgao committed:
Merge pull request #7375 from cBioPortal/jjgao-patch-4-1

Fix #7372

inodb committed:
Merge pull request #7443 from kalletlak/fix-7418

Temporary fix for error in mutation enrichments

inodb committed:
Publish tags to docker hub

Fix #7442. Publish docker images for each tag on github

Note that we can't reuse the existing master image as proposed in the issue, because the version in the pom is determined on the fly from git i.e. the latest master version shows a different version in the api/info response than a proper tagged version even though the code might be completely the same otherwise.

inodb committed:
Merge pull request #7456 from cBioPortal/publish-tag-github-action-docker

Publish tags to docker hub

dippindots committed:
do not update generic assay by default, do not delete entities if they are shared with other studies

dippindots committed:
update function
